src/main/scala/com/gradle/internal/CustomBuildScanEnhancements.scala

```diff
@@ -1,328 +1,369 @@
+      captureTeamCity,
+      captureCircleCi,
+      captureBamboo,
+      captureGitHubActions,
+      captureGitLab,
+      captureTravis,
+      captureBitrise,
+      captureGoCd,
+      captureAzurePipelines,
+      captureBuildkite
+    )
+    Function.chain(ops)
+  }
+
+  private val captureGitMetadata: BuildScan => BuildScan = {
+    if (!isGitInstalled) identity
     else {
-      if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-        val webRepoUri = toWebRepoUri(gitRepo.get)
-        webRepoUri.foreach(uri => {
-          if (uri.getHost.contains("github")) {
-            buildScan.link("GitHub source", uri + "/tree/" + gitCommitId.get)
-          }
-          else {
-            if (uri.getHost.contains("gitlab")) {
-              buildScan.link("GitLab source", uri + "/-/commit/" + gitCommitId.get)
-            }
-          }
-        })
+      val gitRepo = Utils.execAndGetStdOut("git", "config", "--get", "remote.origin.url")
+      val gitCommitId = Utils.execAndGetStdOut("git", "rev-parse", "--verify", "HEAD")
+      val gitCommitShortId = Utils.execAndGetStdOut("git", "rev-parse", "--short=8", "--verify", "HEAD")
+      val gitBranchName = getGitBranchName()
+      val gitStatus = Utils.execAndGetStdOut("git", "status", "--porcelain")
+      val githubRepositoryLink = for {
+        githubUrl <- env.envVariable[URL]("GITHUB_SERVER_URL")
+        repository <- env.envVariable[String]("GITHUB_REPOSITORY")
+        commit <- gitCommitId
+      } yield ("GitHub source", sbt.url(s"$githubUrl/$repository/tree/$commit"))
+      lazy val webRepo = for {
+        origin <- gitRepo
+        commit <- gitCommitId
+        webRepoUri <- Utils.toWebRepoUri(origin)
+        host = webRepoUri.getHost
+        if host.contains("github") || host.contains("gitlab")
+      } yield {
+        if (host.contains("github")) ("GitHub source", sbt.url(s"$webRepoUri/tree/$commit"))
+        else ("GitLab source", sbt.url(s"$webRepoUri/-/commit/$commit"))
       }
+
+      val ops = Seq(
+        ifDefined(gitRepo)((bs, repo) => bs.value("Git repository", Utils.redactUserInfo(repo))),
+        ifDefined(gitCommitId)(_.value("Git commit id", _)),
+        ifDefined(gitCommitShortId)(withCustomValueAndSearchLink(_, "Git commit id", "Git commit id short", _)),
+        ifDefined(gitBranchName) { (bs, branch) => bs.tag(branch).value("Git branch", branch) },
+        ifDefined(gitStatus)(_.tag("Dirty").value("Git status", _)),
+        ifDefined(githubRepositoryLink.orElse(webRepo)) { case (bs, (label, uri)) => bs.link(label, uri) }
+      )
+      Function.chain(ops)
     }
   }
 
   private lazy val isGitInstalled: Boolean = {
-    val installed = execAndCheckSuccess("git", "--version")
-    if (!installed) println("[info] Git executable missing")
+    val installed = Utils.execAndCheckSuccess("git", "--version")
+    if (!installed) logger.info("Git executable missing")
     installed
   }
 
   private def getGitBranchName(): Option[String] = {
```

Comment:
not related to this PR, but @ribafish can you align this function with the newest functionality in the CCUD Gradle plugin?

Reply:
Will do after this is merged